hammers/query.py

```diff
@@ -403,21 +403,45 @@ def update_orphan_resource_providers(db):
     '''
     return db.query(sql, no_rows=True)
 
+
+@query
+def get_nodes_by_lease(db, lease_id):
+    """Get node ids of hosts in a lease."""
+    sql = '''\
+    SELECT ch.hypervisor_hostname AS node_id
+    FROM blazar.leases AS l
+    JOIN blazar.reservations AS r ON l.id = r.lease_id
+    JOIN blazar.computehost_allocations AS ca ON ca.reservation_id = r.id
+    JOIN blazar.computehosts AS ch ON ca.compute_host_id = ch.id
+    WHERE l.id = %s
+    '''
+
+    return db.query(sql, (lease_id,), limit=None)
+
+
 @query
 def get_advance_reservations(db):
     """Get all advance reservations created by any user"""
     sql = '''\
-    SELECT lu.name AS user_name, u.extra AS user_extra, l.name AS lease_name, l.id AS lease_id, p.name AS project_name, l.start_date AS start_date
+    SELECT lu.name AS user_name
+        , u.extra AS user_extra
+        , l.name AS lease_name
+        , l.id AS lease_id
+        , p.name AS project_name
+        , l.start_date AS start_date
     FROM blazar.leases AS l
     JOIN keystone.user AS u ON l.user_id = u.id
     JOIN keystone.local_user AS lu ON u.id = lu.user_id
     JOIN keystone.project AS p ON l.project_id = p.id
     WHERE l.start_date > UTC_TIMESTAMP()
         AND l.deleted_at IS NULL
+        AND p.name NOT LIKE "Chameleon%"
```

Comment:
Two suggestions:

1) Make this use `p.id`, which cannot change. Name can and does change (with project nicknames). Bonus, you can do `p.id IN (some_id, other_id)`
2) I would perform this type of filtering post-DB. It is a bit easier to customize filter logic in code rather than in a database query. But this works just as well, really. If you wanted to filter by project ID in code, the downside is you'd have to pull `p.id as project_id` and keep track of it somewhere. So actually, maybe your approach is better :)

Reply:
Moved exclusion list to lease_stack_reaper and added exogeni@chameleon project to list.